app/coffee/RoomManager.coffee

```diff
@@ -113,3 +112,9 @@ module.exports = RoomManager =
     # calling it asynchronously would lead to race conditions -- see above
     getClientsInRoomPseudoAsync: (io, room, cb) ->
         cb(null, RoomManager.getClientsInRoomSync(io, room))
+
+    getRoomsClientIsInSync: (client) ->
+        roomRegistry = client.server.sockets.adapter.sids
+        return [] unless roomRegistry.hasOwnProperty(client.id)
+        # skip the socket id
+        return Object.keys(roomRegistry[client.id]).slice(1)
```

Comment:
I see it was like this already but is there a way to exclude the client's own socket robustly, rather than depending on its position in the array returned by `Object.keys`?  I am not sure whether the order is guaranteed -- looking at MDN (which may not be 100% reliable) says `Object.keys` returns the same order as `for...in`, and that `for...in` order is implementation-dependent.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in

Reply:
The id is on first position in Node 12 and 14 too -- but I can make it more explicit.